Add custom anchor to download-section.html, and use it to fix link on previews download page

JimmyCushnie opened the pull request:
Alternative to #999.

There's a link at the top of the [previews download page](https://godotengine.org/download/preview/) which says "Looking for a stable version? See below!"

This link is supposed to scroll the page to the section lower down where you can download the stable version. However, currently this link doesn't work because it points to an anchor that no longer exists.

The old anchor, `#stable`, was [removed](https://github.com/godotengine/godot-website/commit/bb0f0a41c3cb1a69dbf7f96ce9c4d98eb9e55822#diff-48f02e6e05568e0d339b6cce00e810761d7046d12d88c855796f6c19ac6269b9L127) in https://github.com/godotengine/godot-website/commit/bb0f0a41c3cb1a69dbf7f96ce9c4d98eb9e55822, when `download-section.html` was consolidated.

This PR lets you set a custom anchor when you include `download-section.html`, and does so in `preview.html` to restore the broken link.

JohnVeness commented:
I have tested locally and this works great. The only other page that includes download-section.html is features.html, and that still works fine (still uses `#call-to-action`, although as far as I can tell nothing internal links to that anyway).

@coppolaemilio Do you want to check this?

coppolaemilio commented on _includes/download/download-section.html:
```diff
@@ -57,7 +57,14 @@
-<div id="call-to-action" class="container card section-download">
+<div 
+    {% if include.anchor %}
+        id="{{ include.anchor }}"
```
I believe there was a way to set up a default value if a variable is not found. I can look it up later on, or if you are able to find it it will simplify this into just one line.

JimmyCushnie replied:
Thanks very much! I used this feature for the anchor as you suggested :) I also noticed it could be used for the header right nearby, so I went ahead and did it there as well.

JimmyCushnie committed:
Add custom anchor to download-section.html, and use it to fix link on preview.html

There's a link at the top of the previews download page which says "Looking for a stable version? See below!"

This link is supposed to scroll the page to the section lower down where you can download the stable version. However, prior to this commit, the link doesn't work because it points to an anchor that no longer exists.

The link points to #stable. However, the actual anchor, as defined in download-section.html, is #call-to-action.

This commit lets you set a custom anchor when you include download-section.html, and does so in preview.html to fix the broken link.

JimmyCushnie committed:
Simplify header in download-section.html

coppolaemilio commented:
Awesome! thanks :)

JimmyCushnie commented:
Woohoo! Happy PR #1000 everyone!
